packages/dd-trace/src/exporters/common/request.js

```diff
@@ -34,40 +37,45 @@ function request (data, options, keepAlive, callback) {
     options.headers['Datadog-Container-ID'] = containerId
   }
 
-  if (keepAlive) {
-    options.agent = isSecure ? httpsAgent : httpAgent
-  }
-
-  const onResponse = res => {
-    let responseData = ''
-
-    res.setTimeout(timeout)
-
-    res.on('data', chunk => { responseData += chunk })
-    res.on('end', () => {
-      activeRequests--
-
-      if (res.statusCode >= 200 && res.statusCode <= 299) {
-        callback(null, responseData, res.statusCode)
-      } else {
-        const error = new Error(`Error from the endpoint: ${res.statusCode} ${http.STATUS_CODES[res.statusCode]}`)
-        error.status = res.statusCode
-
-        callback(error, null, res.statusCode)
-      }
-    })
-  }
+  options.agent = isSecure ? httpsAgent : httpAgent
 
   const makeRequest = onError => {
-    if (!request.writable) return callback(null)
+    if (!request.writable) {
+      log.debug('Payload discarded')
+      return callback(null)
+    }
 
     activeRequests++
 
     const store = storage.getStore()
 
     storage.enterWith({ noop: true })
 
-    const req = client.request(options, onResponse)
+    const req = client.request(options, res => {
+      let responseData = ''
+
+      res.setTimeout(timeout)
+
+      res.on('data', chunk => { responseData += chunk })
+      res.on('end', () => {
+        if (req.isTimedOut) {
```

Comment:
You are not reducing `activeRequest` on timeout, should you change it to this?
```suggestion
        activeRequests--
        if (req.isTimedOut) {
          return
        }

Reply:
Ok, I have realized now that you are doing `req.destroy(new Error('Request timed out'))` and it would probably launch an `error` event, that reduces `activeRequests`

Reply:
that's right 👍